File: 01.JS-Basic/ИЗПИТ/Everest.js
function everest(input) {

    let currentScore = 5364
    let days = 1
    let i = 0

    while (currentScore < 8848) {

        let currentCheck = input[i]

        if (currentCheck === "No") {
            i++
            currentScore += Number(input[i])
        } else if (currentCheck === 'Yes') {
            days++
            if (days > 5) {
                break;
            } else {
                i++
                currentScore += Number(input[i])
            }
        } else if (currentCheck === 'END') {
            break;
        }
        i++
    }
    if (currentScore < 8848) {
        console.log("Failed!");
        console.log(currentScore);
    } else {
        console.log(`Goal reached for ${days} days!`);
    }
}
everest(['Yes',
    '535',
    'Yes',
    '849',
    'Yes',
    '499',
    'Yes',
    '400',
    'Yes',
    '500'])

File: JAVA-Basic/JavaBasic/src/IZPIT/Everest.java
package IZPIT;

import java.util.Scanner;

public class Everest {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int currentScore = 5364;
        int days = 1;
        int i = 0;
        String currentCheck = scan.nextLine();

        while (!currentCheck.equals("END")) {
            int currentDistance = 0;

            if (currentCheck.equals("No")) {
                currentDistance = Integer.parseInt(scan.nextLine());
                currentScore += currentDistance;
            } else if (currentCheck.equals("Yes")) {
                days++;
                if (days > 5) {
                    break;
                } else {
                    currentDistance = Integer.parseInt(scan.nextLine());
                    currentScore += currentDistance;
                }
            }
            if (currentScore >= 8848) {
                break;
            }
            currentCheck = scan.nextLine();
        }
        if (currentScore < 8848) {
            System.out.println("Failed!");
            System.out.println(currentScore);
        } else {
            System.out.printf("Goal reached for %d days!", days);
        }
    }
}
